config/render.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// RenderApp encapsulates the rendered configuration file and the input and
+// output paths.
+type RenderApp struct {
+	renderFlag     string
+	renderedConfig []byte
+}
+
+// EmptyApp creates an empty application
+func EmptyApp() *RenderApp {
+	app := &RenderApp{}
+	return app
+}
+
+// NewRenderApp creates a new Config Rendering application
+func NewRenderApp(configFlag string, renderFlag string) (*RenderApp, error) {
```

Comment:
Oh actually maybe an even better signature would be `func (c *Config) Render(io.Writer)`, so that the caller can pass in an arbitrary `Writer` for us to stream bytes into.